src/meta/service/src/ddl_service.rs

```diff
@@ -1095,21 +1033,11 @@ impl DdlServiceImpl {
             .catalog_controller
             .get_connection_by_id(connection_id as _)
             .await?;
-        if let Some(connection::Info::PrivateLinkService(svc)) = &connection.info {
-            // skip all checks for mock connection
-            if svc.get_provider()? == PrivateLinkProvider::Mock {
-                return Ok(());
-            }
-
-            // check whether private link is ready
-            if let Some(aws_cli) = self.aws_client.as_ref() {
-                if !aws_cli.is_vpc_endpoint_ready(&svc.endpoint_id).await? {
-                    return Err(MetaError::from(anyhow!(
-                        "Private link endpoint {} is not ready",
-                        svc.endpoint_id
-                    )));
-                }
-            }
+        if let Some(connection::Info::PrivateLinkService(_)) = &connection.info {
```

Comment:
Here is in `validate_connection`. Will it be called during startup or recovery? If possible, we should allow an existing PrivateLink connection to work after upgrading RW.

Reply:
> Will it be called during startup or recovery

Nope. We only call `validate_connection` when create_source and create_sink, not blocking recovery.